### PR TITLE
use scratchpad for IW working copy

### PR DESCRIPTION
**Pull request by amunger**
- Remove the Interactive file scheme and FS provider as they will now be untitled files
- Use the notebook serializer for the IW to simplify saving as an ipynb
  - `save as` will save and re-open as an ipynb
- Stop registering the notebook editor for `*.interactive` to prevent overwriting the interactive contributions
- Use untyped input for `editorService.OpenEditor` to adopt the editor resolver, rather than creating the `editorInput` ourselves
  - required for letting the WCS handle backup recovery

### CODE REVIEW

File: extensions/ipynb/package.json
Comment (bpasero) [inline extensions/ipynb/package.json:18]: Formatting issue?

**Comment by bpasero** [comment]
I am not deep in any of the code of this PR, is there something specifically you want me to review here? Otherwise I would yield for Pung.

**Comment by amunger** [comment]
@bpasero - mostly the change in editor registration and usage of the editor service resolver, though I think I have it pretty figured out.

File: src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts:283]: I wonder if we can have better name to indicate that we only want to register the notebook provider info (for resolving notebook model) but we don't want to register an editor.

**Review by rebornix — APPROVED** [review]

**Comment by amunger** [comment]
Actually @bpasero, there is a remaining issue that I will probably want your review on, but I didn't manage to get it done today:

When closing without hot exit:
- Reverting doesn't drop the IW backup. I think that's because we don't fire onDidChangeDirty for scratchpads (so might end up needing a new event pipeline)
- Saving doesn't save the editor in Editor.SaveAll because it only looks for dirty editors to save. It saves on the working copy instead, which doesn't use the suggested target, and also leaves the old editor tab open.

**Comment by bpasero** [comment]
@amunger is that something for a follow up issue or blocking this PR? if its not blocking then maybe we can do that in a new issue to get this change rolling?

**Comment by amunger** [comment]
it would definitely make sense to put it in a separate change, so I'll follow up with that fix after this is merged

**Pull request merged**
